source/simulator/stokes_matrix_free.cc

```diff
@@ -885,17 +885,23 @@ namespace aspect
 
     this->set_constrained_entries_to_one(diagonal);
     inverse_diagonal = diagonal;
-    const unsigned int local_size = inverse_diagonal.local_size();
-    for (unsigned int i=0; i<local_size; ++i)
+
+    // Finally loop over all of the computed diagonal elements and invert them.
+    // The following loop relies on the fact that inverse_diagonal.begin()/end()
+    // iterates only over the *locally owned* elements of the vector in which
+    // we store inverse_diagonal.
+    for (auto local_element : inverse_diagonal)
```

Comment:
Even though C++11 is old by now, I am still unsure about how to use `auto` correctly in some situations. But does this need to be an `auto &local_element` to make the assignment below actually change anything?

Reply:
Good question. Whatever this does, it seems to result in an lvalue that I can assign to a couple of lines below. But it is questionable for sure, so let's try the thing with the reference :-)